apps/angular/crud/src/app/app.component.ts

```diff
@@ -1,51 +1,58 @@
 import { CommonModule } from '@angular/common';
-import { HttpClient } from '@angular/common/http';
-import { Component, OnInit } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnInit,
+  signal,
+} from '@angular/core';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
 import { randText } from '@ngneat/falso';
+import { AppService } from './app.service';
+import { User } from './appInterface.component';
 
 @Component({
   standalone: true,
-  imports: [CommonModule],
+  imports: [CommonModule, MatProgressSpinnerModule],
+  providers: [AppService],
   selector: 'app-root',
   template: `
-    <div *ngFor="let todo of todos">
-      {{ todo.title }}
-      <button (click)="update(todo)">Update</button>
-    </div>
+    @for (todo of todos(); track todo.id) {
+      <div>
+        {{ todo.title }}
+        <button (click)="update(todo)">Update</button>
+        <button (click)="deleteUsers(todo.id)">Delete</button>
+      </div>
+    } @empty {
+      <mat-spinner></mat-spinner>
+    }
   `,
   styles: [],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class AppComponent implements OnInit {
-  todos!: any[];
+  todos = signal<User[]>([]);
 
-  constructor(private http: HttpClient) {}
+  constructor(private appService: AppService) {}
 
   ngOnInit(): void {
-    this.http
-      .get<any[]>('https://jsonplaceholder.typicode.com/todos')
-      .subscribe((todos) => {
-        this.todos = todos;
-      });
+    this.appService.getAllUsers().subscribe((todos) => {
+      this.todos.set(todos);
+    });
   }
 
-  update(todo: any) {
-    this.http
-      .put<any>(
-        `https://jsonplaceholder.typicode.com/todos/${todo.id}`,
-        JSON.stringify({
-          todo: todo.id,
-          title: randText(),
-          body: todo.body,
-          userId: todo.userId,
-        }),
-        {
-          headers: {
-            'Content-type': 'application/json; charset=UTF-8',
-          },
-        },
-      )
-      .subscribe((todoUpdated: any) => {
-        this.todos[todoUpdated.id - 1] = todoUpdated;
-      });
+  update(todo: User) {
+    const todoClone = { ...todo };
+    todoClone.title = randText();
+    this.appService.updateUser(todoClone).subscribe((todoUpdated: User) => {
```

Comment:
You need to be careful with subscribing. I always prefer to add an unsubscribe method even if I know that http request complete. You never know what the network will be.